src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -143,6 +144,40 @@ pub(super) struct Peripherals {
     pub mmio_bus: Option<crate::devices::Bus>,
 }
 
+/// If any of per-vCPU IA32_TSC_DEADLINE MSRs are zero, update them
+/// with the maximum value observed across all vCPUs.
+///
+/// We observed that sometimes when taking a snapshot,
+/// the IA32_TSC_DEADLINE MSR is cleared, but the interrupt is not
+/// delivered to the guest, leading to a situation where one
+/// of the vCPUs never receives TSC interrupts after restoring,
+/// until the MSR is updated externally, eg by setting the system time.
+pub fn fix_zero_tsc_deadline_msr(vcpu_states: &mut [VcpuState]) {
+    let max_tsc_deadline_val = vcpu_states
```

Comment:
Is there any reason to use the max value? Doesn't setting it to 1 (the minimum number larger than 0) work? If the interrupt just gets ignored, it makes sense to me to set vCPU interrupted as soon as possible. wdyt?

Reply:
That's quite right, I didn't think about it. Looks like a better option!

Reply:
done